From d5bc4ad53a2bee73946c6977870316e19d848680 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v11-07.12-d5bc4ad53a2-20220215T235428Z-avarab@gmail.com>
In-Reply-To: <cover-v11-00.12-00000000000-20220215T235428Z-avarab@gmail.com>
References: <cover-v10-0.9-00000000000-20220204T211534Z-avarab@gmail.com>
	<cover-v11-00.12-00000000000-20220215T235428Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 4 Feb 2021 16:33:09 +0100
Subject: [PATCH v11 07/12] Revert "wip reg_newline etc. test"

This reverts commit 3d49287f57d1c02241b52a41c0e9b143cabb590e.
---
 grep.c                 |  2 +-
 t/t4209-log-pickaxe.sh | 10 ++--------
 2 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/grep.c b/grep.c
index 0a27a81bb4f..951d0faeb93 100644
--- a/grep.c
+++ b/grep.c
@@ -430,7 +430,7 @@ static void compile_fixed_regexp(struct grep_pat *p, struct grep_opt *opt)
 static void compile_regexp(struct grep_pat *p, struct grep_opt *opt)
 {
 	int err;
-	int regflags = 0;
+	int regflags = 0;//REG_NEWLINE
 
 	if (opt->pattern_type_option == GREP_PATTERN_TYPE_UNSPECIFIED)
 		opt->pattern_type_option = (opt->extended_regexp_option
diff --git a/t/t4209-log-pickaxe.sh b/t/t4209-log-pickaxe.sh
index 91c622175a2..f1f11042838 100755
--- a/t/t4209-log-pickaxe.sh
+++ b/t/t4209-log-pickaxe.sh
@@ -149,7 +149,7 @@ test_expect_success 'log -S --no-textconv (missing textconv tool)' '
 test_expect_success 'setup log -[GS] plain & regex' '
 	test_create_repo GS-plain &&
 	test_commit -C GS-plain --append A data.txt "a" &&
-	test_commit -C GS-plain --append B data.txt "aa" &&
+	test_commit -C GS-plain --append B data.txt "a a" &&
 	test_commit -C GS-plain --append C data.txt "b" &&
 	test_commit -C GS-plain --append D data.txt "[b]" &&
 	test_commit -C GS-plain E data.txt "" &&
@@ -158,7 +158,6 @@ test_expect_success 'setup log -[GS] plain & regex' '
 	git -C GS-plain log --grep="[ABE]" >A-to-B-then-E-log &&
 	git -C GS-plain log --grep="[CDE]" >C-to-D-then-E-log &&
 	git -C GS-plain log --grep="[DE]" >D-then-E-log &&
-	git -C GS-plain log --grep="[AE]" >A-then-E-log &&
 	git -C GS-plain log >full-log
 '
 
@@ -166,12 +165,7 @@ test_expect_success 'log -G trims diff new/old [-+]' '
 	git -C GS-plain log -G"[+-]a" >log &&
 	test_must_be_empty log &&
 	git -C GS-plain log -G"^a" >log &&
-	test_cmp A-to-B-then-E-log log
-'
-
-test_expect_success 'log -S --pickaxe-regex uses REG_NOTBOL for subsequent matches' '
-	git -C GS-plain log -S"^a" --pickaxe-regex >log &&
-	test_cmp A-then-E-log log
+	test_cmp log A-to-B-then-E-log
 '
 
 test_expect_success 'log -S<pat> is not a regex, but -S<pat> --pickaxe-regex is' '
-- 
2.35.1.1027.g942e091e858

